### PR TITLE
Level 1/task 4

### PR DESCRIPTION
**Pull request by PolinaXox**
The inspiration for the status codes: [the last  image on the page](https://auth0.com/blog/forbidden-unauthorized-http-status-codes/)

### CODE REVIEW

File: 1. Basics/task_4/hw4.php [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        
+        if($body['login'] == $login) {
+            if($body['password'] == $password) {
+                
+                // login found, password matched
+                return 200;
+            }
+
+            // login found, but password DIDN'T match
+            else
```
Comment (anton-moroz) [inline 1. Basics/task_4/hw4.php:241]: is `else` there really needed?

File: 1. Basics/task_4/hw4.php [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    if(isset(RESPONSE_DATA[$statusCode]['color'])) {
+        $body .= ' style="color:' . RESPONSE_DATA[$statusCode]['color'] . '"';
+    }
+
+    $body .= ">";
+    
+    if(isset(RESPONSE_DATA[$statusCode]['bodyMessage'])) {
+        $body .= RESPONSE_DATA[$statusCode]['bodyMessage'];
+    }
+    else {
```
Comment (anton-moroz) [inline 1. Basics/task_4/hw4.php:308]: use ternary operator

File: 1. Basics/task_4/hw4.php [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+ * @param string $delim2
+ * 
+ * @return array
+ */
+function getStringContentAsPairsArray($str, $delim1, $delim2) : array {
+    $arr = cleanArray(explode($delim1, $str));
+    $pairsAsArr = [];
+    
+    foreach($arr as $pair) {
+        $p = explode($delim2, $pair);subject:
```
Comment (anton-moroz) [inline 1. Basics/task_4/hw4.php:273]: what is `subject:`?

**Review by anton-moroz — CHANGES_REQUESTED** [review]

File: 1. Basics/task_4/hw4.php (hunk 3 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php:273]: image
Its color tells that is the part on my new debugger using result. Now I have no ideas how it became the part of my code.
I'll remove it.

File: 1. Basics/task_4/hw4.php (hunk 1 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php:241]: I think "yes in this option". It prevets to read all the long login-password associative array. Most resources do not allow multiple users with the same names(login), so if the login was found and the password was wrong, this ```else break;``` says: "Stop! No need to search more." and breaks the outer loop too.

The other options: resource allow the scheme "one login - some passwords", logins are sorted and intelligent search uses, logins and passwords store in the hash-map.

File: 1. Basics/task_4/hw4.php (hunk 2 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php]: :ok: Done

File: 1. Basics/task_4/hw4.php (hunk 1 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php]: Concept has been was changed. Now there is an operator ```throw``` instead.

File: 1. Basics/task_4/hw4.php [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+    
+    // if the request has headers store each header data as
+    // $headers[headerName] = headerValue
+    while($i < $arrSize) {
+        $ind = strpos($strAsArr[$i], ":");
+        $headers[trim(string: substr($strAsArr[$i], 0, $ind))] = trim(substr($strAsArr[$i], $ind+1));
+        $i++;
+    }
+    
+    return array(
```
Comment (anton-moroz) [inline 1. Basics/task_4/hw4.php:115]: use `[]`

File: 1. Basics/task_4/hw4.php [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+ * @param string $delim2
+ * 
+ * @return array
+ */
+function getStringContentAsPairsArray($str, $delim1, $delim2) : array {
+    $arr = cleanArray(explode($delim1, $str));
+    $pairsAsArr = [];
+    
+    foreach($arr as $pair) {
+        $p = explode($delim2, $pair);subject:
```
Comment (anton-moroz) [inline 1. Basics/task_4/hw4.php:201]: remove `subject:`

File: 1. Basics/task_4/hw4.php [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+    throw new Exception('File ' . $file . ' doesn`t exist', 500);
+}
+
+/**
+ * Grants or denies access. Returns the result code and message.
+ * @param array $body
+ * @throws Exception
+ * @return array
+ */
+function getAccess($body) : array {
```
Comment (anton-moroz) [inline 1. Basics/task_4/hw4.php:316]: not obvious what this function is doing, should be renamed

File: 1. Basics/task_4/hw4.php [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+    $logAndPassPairsAsArr = getFileContentAsPairsArray(FILE_NAME, PHP_EOL, ":");
+
+    foreach($logAndPassPairsAsArr as $login => $password) {
+        if($body['login'] == $login) {
+            if($body['password'] == $password) {
```
Comment (anton-moroz) [inline 1. Basics/task_4/hw4.php:321]: remove `else`

Comment (anton-moroz) [reply 1. Basics/task_4/hw4.php:321]: leave just `throw Exception`

**Review by anton-moroz — CHANGES_REQUESTED** [review]

File: 1. Basics/task_4/hw4.php (hunk 7 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php:321]: Done. Exactly done. Checked. 💯

File: 1. Basics/task_4/hw4.php (hunk 6 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php]: Done. Now its name is ```checkLoginAndPassword```

File: 1. Basics/task_4/hw4.php (hunk 5 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php]: Done. Exactly done. Checked. 💯

File: 1. Basics/task_4/hw4.php (hunk 4 above)
Comment (PolinaXox) [reply 1. Basics/task_4/hw4.php]: Done. Exactly done. Checked. 💯

**Review by anton-moroz — APPROVED** [review]

**Pull request merged**